src/components/pages/AnnotationPage.js

```diff
@@ -0,0 +1,25 @@
+import React, {Component}  from "react";
+import { connect } from "react-redux";
+import { Link, useParams } from "react-router-dom";
+import { fetchAnnotationDetails } from "../../actions";
+import ImageAnnotation from "../ImageAnnotation";
+
+function AnnotationPage (props) {
+  let { id } = useParams();
+  const { annotation } = props;
+  if (!annotation || annotation._id != id) {
```

Comment:
if annotation is empty we need to load it. Or if annotation id is different to annotation id from the url this means that we need to reload it.